nim_status/multiaccount.nim

```diff
@@ -49,27 +59,45 @@ proc buildAccount(privateKey: PrivateKey): Account =
 
   return acc
 
-proc generateAndDeriveAddresses*(mnemonicPhraseLength: int, n: int,
-  bip39Passphrase: string, paths: seq[string]): seq[MultiAccount] =
+proc deriveAccounts*(multiAcc: MultiAccount, paths: seq[KeyPath]): seq[Account] =
+  var accounts: seq[Account]
+  for p in paths:
+    let skResult = derive(multiAcc.keySeed, p)
+    var acc = buildAccount(cast[PrivateKey](skResult.get()))
+    acc.path = p
+    accounts.add(acc)
+  return accounts
+
+proc importMnemonic*(mnemonicPhrase: Mnemonic, bip39Passphrase: string): MultiAccount =
+  let seed = getSeed(Mnemonic mnemonicPhrase, bip39Passphrase)
+  # Ensure seed is within expected limits
+  let lseed = len(cast[seq[byte]](seed))
```

Comment:
why the cast here? it doesn't do anything - also, casting to seq should be avoided, it's generally not safe  (even if there exist a few exceptional cases)